### PR TITLE
Custom commands, environment switching and local sandbox

### LINKED ISSUE #17: Support custom commands
## Acceptance Criteria

1. A project can define its own commands that are registered and made available by the tool

Child of #19

### LINKED ISSUE #18: Implementation of environment switch and deployment scripts
Fill out after completing #2

Child of #16

### LINKED ISSUE #26: Local sandbox command
Child of #16

I want to pull a site database and run deployment commands, so that I can have a local sandbox that mirrors production.

In ahoy, this is accomplished with the site up command. 

Acceptance criteria:
- Project configuration tells command where to find database
- imports database
- download the files

### PR DESCRIPTION
**Pull request by fmizzell**
## Custom commands

Fixes #17 
**QA**
- Run 'dktl init'
- Create the class given in the README.md into src/command/CustomCommands.php
- Run 'dktl custom:sample'
- Should see "Hello World!!" printed

## Local sandbox

Fixes #26

`dktl dkan:restore`

Also adds ability to call dktl within CLI container, so that we can for instance call `_exec('dktl whatever')` in a command and it will work.
**QA Steps**

1. New folder, `dktl dc up -d`
2. `dktl init`
3. `dktl dkan:get 7.x-1.15.2 && dktl dkan:make && dktl drupal:make`
4. `dktl dkan:restore https://s3.amazonaws.com/dkan-tools/last_install.sql https://s3.amazonaws.com/dkan-tools/files.zip`

## Deployment commands

Fixes #18 

1. run a dkan:restore
2. Run `dktl dkan:deploy local`
3. Check modules to make sure devel module enabled
4. Admin bar should show local environment

### CODE REVIEW

**Comment by kimwdavidson** [comment]
@fmizzell should this be attached to something? or in code review?

File: assets/command/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+## DKAN Tools Custom Commands
```
Comment (dafeder) [inline assets/command/README.md]: Remove this file from assewts

File: src/Util/ArgumentsAndOptionsLoader.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<?php
```
Comment (dafeder) [inline src/Util/ArgumentsAndOptionsLoader.php]: Rename class to ArgumentLoader

File: src/Util/ArgumentsAndOptionsLoader.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public function enhancedArgv() {
+        $argv = $this->argv;
+        $command = $this->getCommand();
+
+        if (!empty($command) && $this->onlyCommandGiven()) {
+            $yamld_command = str_replace(":", "|", $command);
+            $config = Yaml::parse(file_get_contents("/var/www/dktl.yml"));
+            print_r($config);
```
Comment (dafeder) [inline src/Util/ArgumentsAndOptionsLoader.php]: Remove debugging

File: src/Util/ArgumentsAndOptionsLoader.php [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public function getCommand() {
+        $command = "";
+        if (count($this->argv) >= 2) {
+            print_r($this->argv);
+            $command = $this->argv[1];
+        }
+        return $command;
+    }
+
+    public function enhancedArgv() {
```
Comment (dafeder) [inline src/Util/ArgumentsAndOptionsLoader.php]: new method name: `getAlteredArgv()`

**Pull request merged**